Modules/rat_board.py

```diff
@@ -199,10 +199,50 @@ def find_by_uuid(self, guid: UUID) -> Optional[Rescue]:
 
         """
         for rescue in self.rescues.values():
-            if rescue.case_id == guid:
+            if rescue.uuid == guid:
                 return rescue
         return None
 
+    def search(self, case: Union[str, int, UUID]) -> Optional[Rescue]:
+        """
+        Helper for searching for a case using implemented search functions.
+
+        Args:
+            case (Union[str, int, UUID]): the client's name, a case ID, or a rescue UUID
+
+        Returns:
+            Rescue: found rescue
+            None:   No rescue found
+
+        Raises:
+            ValueError: argument was not an expected type
+        """
+        # ensure a proper type was passed
+        if not isinstance(case, str) and not isinstance(case, int) and not isinstance(case, UUID):
+            raise ValueError
+
+        if isinstance(case, str):
+            # strings could be any of the valid types, check for the best one.
+            if case.isdigit():
+                    return self.find_by_index(int(case))
+            else:
+                # UUIDs always start with @ and are 37 characters, so try to parse it
+                if case[0] == "@" and len(case) == 37:
+                    try:
+                        guid_str = case[1:]
+                        guid = UUID(guid_str)
+                    except (AttributeError, ValueError):
+                        # not a valid UUID
+                        return None
+                    else:
+                        return self.find_by_uuid(guid)
+                else:
+                    return self.find_by_name(case)
+        elif isinstance(case, int) and not isinstance(case, bool):
```

Comment:
remind me, why are we considering booleans here?

Reply:
Python assumes True/False are also ints, so the isinstance(case, int) will return True. Perhaps this check is better served in the first assert statement.

Reply:
True/false are singletones in Python, it may be better to assert if the value /is/ a boolean rather than an instance thereof.